[ADD] timesheets: new doc timesheets assistant (#17381)
odoo/documentation: content/applications/services/timesheets/timesheets_assistant.rst

====================
Timesheets assistant
====================

Timesheets Assistant monitors the user’s activity on their computer and in their Internet browser to

> simplify timesheet logging.

Reviewer: I suggest replacing the whole intro with something like this (see my comments below for the why):

```suggestion
The timesheet assistant monitors computer and browser activities using `ActivityWatch <https://activitywatch.net/>`_ to generate timesheet entry suggestions and simplify timesheet logging. To use it, ref:`activate the feature <timesheets/timesheets-assistant/activation>` and :ref:`install the ActivityWatch app <timesheets/timesheets-assistant/install-desktop>` on the computer, along with the :ref:`browser extension
 <timesheets/timesheets-assistant/install-browser>` if browser activity tracking is required. Suggestions are generated based on :ref:`assistant rules <timesheets/timesheets-assistant/assistant-rules>` and calendar events and planning in Odoo, then mapped to projects and tasks, taking into account previously matched suggestions. They can then be :ref:`reviewed and converted into timesheet entries <timesheets/timesheets-assistant/use>` directly from the timesheet assistant view.
```
don't forget to add the timesheets/timesheets-assistant/activation anchor above the related section

The calculation is based on predefined rules and learning from the user’s previous matches between
suggested timesheets and their assigned projects. These rules act as *key events*. Once a key event
is triggered, any subsequent activity of minimum one minute is associated with that event until a
new key event is triggered. The more rules that are configured, the more precise the results become
and the less they are affected by short or irrelevant entries.

By default, Timesheets Assistant includes ready-to-use rules to track the time spent working in
Odoo, on GitHub and Google Docs, browsing Gmail, and using Discord. These rules can be edited or

> removed, and new ones can be added by the user.

Reviewer: I think this information should be simplified for this introduction and the details (for example, everything about "key events" and default rules) moved the the relevant configuration section:

- Key events: I'd include this in the Use + Local assistant rules sections
- Default rules + everything about rules: should be mentionned in the Assistant rules section

--> my suggestion in those sections take this into account and I'd remove the text from here entirely

.. example::
   A user opens a project task in their Odoo database (the key event), then spends one hour browsing
   websites and using various applications until they open another project task. Timesheets
   Assistant then suggests a one-hour timesheet entry linked to the most relevant project. This
   suggestion can be confirmed by matching it, or another project can be selected to associate the
   time with.

Timesheets Assistant also checks the user's calendar events and planning in Odoo to make
suggestions. For instance, if the user has a meeting with a customer from 10 to 11 AM, they will
receive a Timesheet entry suggestion for it.

Periods of keyboard and mouse inactivity are also tracked. Any inactive period of at least 3 minutes

> is considered away-from-keyboard, and Timesheets Assistant identifies that time as a break.

Reviewer: I incmuded calendar events and planning in my suggestion above, but for the rest, I think this information is a bit too detailed for a general intro, where users might not have the context and knowledge necessary to understand everything here (I know I didn't, I had to go through the configuration + use parts to understand this). I made sure to keep the useful and relevant info, but move it to other parts of this doc; my suggestions throughout the PR take this into account.

.. important::
   The user’s activity data is processed locally on their device and is never shared with anyone,
   either within or outside their organization. Not even Odoo, as a company, can access a user’s
   activity.

> Configuring Timesheets Assistant

Reviewer: ```suggestion
Configuration
```

================================

Activation
----------

Timesheets Assistant must be enabled before use.

.. important::
   Timesheets Assistant is only available when the encoding method is set to Hours / Minutes. Go to
   :menuselection:`Settings --> Timesheets` and make sure the Encoding Method is set to
   :guilabel:`Hours / Minutes`.

Then go to :menuselection:`Settings --> Timesheets` and enable :guilabel:`Timesheets Assistant`.

> Click :guilabel:`Save`.

Reviewer: ```suggestion
.. important::
   The timesheet assistant is only available when the :guilabel:`Encoding method` is set to :guilabel:`Hours / Minutes` in the :guilabel:`Timesheets Settings`. 

To enable the timesheet assistant, go to :menuselection:`Timesheets --> Configuration --> Settings --> Timesheets` and enable :guilabel:`Timesheets Assistant`.
```

Installation
------------

A dedicated open source software called ActivityWatch must be installed on the user’s computer and
browser to track their activity.

> Go to :menuselection:`Timesheets --> Assistant`.

Reviewer: ```suggestion
The timesheet assistant tracks user activity through the `ActivityWatch <https://activitywatch.net/> app`_ (for :ref:`computer activity <timesheets/timesheets-assistant/install-desktop>`) and browser extension (for :ref:`browser activity <timesheets/timesheets-assistant/install-browser>`). To install the timesheet assistant, follow these steps:

#. Go to :menuselection:`Timesheet --> Assistant`.
```
- Intro changes: to be a bit more accurate + improve the style
- I think having a short numbered list with main installation steps would be useful (the remaining list items after this first ones are in my comment below)
- See my comment below about adding the anchors for the references

.. note::
   If your browser requests access to apps and services on this device in a pop-up, click
   :guilabel:`Allow`.

   If you don’t see the prompt, in Google Chrome, click the Tune icon to the left of the address
   bar. Click :guilabel:`Site Settings`. Look for a permission related to “Apps on device” (or
   similar) and set it to :guilabel:`Allow`. Refresh the page.

> Go to :menuselection:`Timesheets --> Assistant --> Start Now`.

Reviewer: ```suggestion
#. If the browser displays a popup requesting permission to access apps and devices on your device, click :guilabel:`Allow`.
#. Follow the installation steps for :ref:`Desktop activity <timesheets/timesheets-assistant/install-desktop>` (mandatory) and :ref:`Browser activity <timesheets/timesheets-assistant/install-browser>` (optional).

.. tip::

   If the popup does not appear in Google Chrome, follow these steps:

   #. Click the tune icon (:guilabel:`View site information`) to the left of the address bar.
   #. Click :guilabel:`Site settings`.
   #. At the bottom of the :guilabel:`Permissions` list, set the :guilabel:`Apps on device` field to :guilabel:`Allow`.
   #. Refresh the Odoo Timesheets page.

   To grant this permission in Mozilla Firefox, refer to the `Mozilla documentation
   <https://support.mozilla.org/en-US/kb/control-personal-device-local-network-permissions-firefox>`_.

```
General comments about your original text:
- Since the previous step (line 58 in my previous suggestion) mentions the menu, we're already there so no need to mention it again.
- [Only menus in menuselection](https://www.odoo.com/documentation/master/contributing/documentation/rst_guidelines.html#menu-selection)

Explanations for suggested changes:
- I think we need to be a bit more precise with "the page" you need to refresh; in your original text, "the page" could refer to the Google Chrome settings and I don't think that's what you meant, hence my suggestion above.
- I assume Google Chrome is the recommended browser, but the timesheet assistant works with Firefox as well so I think we should mention it at the bottom of the tip (like I suggested).
- I think adding a link to the sections Desktop activity and Browser activity in the steps would be useful and improve the transition (and context) of those 2 sections; when I started reading the next section (Desktop activity), I was a bit lost as to where I was supposed to be exactly. Make sure to add the refs above the related sections (`timesheets/timesheets-assistant/install-desktop` above Desktop activity and `timesheets/timesheets-assistant/install-browser` above Browser activity).
- I think it makes sense to mention that desktop activity is mandatory and browser activity is optional here already
- I suggested to leave the step about the "Start now" button for the Desktop activity instructions because that's when you will need to click the button; once you've installed the timesheet assistant for browser activity, the button no longer appears and instead you get a cog icon (I took this into account for my suggestions in the Browser activity sections)
- And should we add a seealso to the Activitywatch doc at the end of this section (just before Desktop activity)? https://docs.activitywatch.net/en/latest/ --> this isn't included in my suggestion above

Desktop activity
~~~~~~~~~~~~~~~~

:guilabel:`Desktop Activity` (required) tracks user activity on their computer by analyzing open
windows and applications. The minimum (oldest supported) versions required are Windows 10, macOS
10.15, and Ubuntu 24 on Linux.

> Select the right operating system for your configuration below and follow the steps:

Reviewer: ```suggestion
To track time spent in applications installed on your computer, install the ActivityWatch application by going to :menuselection:`Timesheets --> Assistant `, clicking :guilabel:`Start Now`, and following the instructions for your operating system below.

.. note::
   The minimum supported versions are Windows 10, macOS 10.15, and Ubuntu 22 or later.
```
- Reworked for general style and accuracy
- Apparently it's supported for Ubuntu 22 (this my computer's OS and I was actually able to test it)
- Used a note for "FYI" info

.. tabs::

   .. tab:: Windows

      #. Odoo automatically detects that your operating system is Windows and offers you the option
         to download the appropriate ActivityWatcher installation package for your configuration by
         clicking :guilabel:`Download for Windows`. You can also download the `Windows installation
         package <https://nightly.odoo.com/aw/windows/activitywatch-odoo_patched-latest-setup.exe>`_.

> 

Reviewer: ```suggestion
     #. In the :guilabel:`Install Timesheets Assistant` window, click :guilabel:`Download for Windows` in the :guilabel:`Desktop Activity` section, then select your OS version.
  
        .. note:: 
           The `Windows installation package can also be downloaded directly   <https://nightly.odoo.com/aw/windows/activitywatch-odoo_patched-latest-setup.exe>`_.
```

      #. Follow the installation instructions. To complete the installation, you need to restart
         your computer or log out. Either option will close all running applications. After
         restarting or logging back in, return to Timesheets Assistant to continue setup :

>          :menuselection:`Timesheets --> Assistant --> Start Now` .

Reviewer: ```suggestion
      #. Install the package and follow the instructions.
      #. Once the installation is finished, restart your computer or log out and log back in.
      #. Access the Odoo database, go to :menuslection:`Timesheets --> Assistant`, and click :guilabel:`Start Now`.
```
- to only keep what is necessary as steps
- splitting the info into several steps for clarity
- RST formatting

      #. To help Timesheets Assistant automatically suggest your timesheet entries, your browser
         needs to be able to communicate with your local ActivityWatch server. Click
         :guilabel:`Open Settings` to access the ActivityWatch website. Under Server Settings,
         locate the CORS field. Copy and paste your Odoo database URL exactly as shown, in quotes

>          and without a slash (/) at the end. Example: `"https://my-company.odoo.com"`

Reviewer: - Before we can get here, we need to click Next twice (same before we can test the connection)
- I'd rephrase to focus on actionable steps and leave unnecessary explanations out (they are implied IMO)
- I'd split between several steps since several actions are required and it's too long for one step (which makes it more difficult to follow)
- Also I know this phrasing comes from the UI but to help the assistant blabla makes it sound a bit like this step is optional and it would work without it (which I don't think is the case)
- Guilabel for UI elements (even outside of Odoo)
- I tested copying the database URL both with the button and manually with Ctrl+C and didn't get a trailing slash in either case, so I don't think we need to mention it?

```suggestion
      #. Click :guilabel:`Next` twice.
      #. In the :guilabel:`Configure CORS` window, copy the Odoo database URL.
      #. Click :guilabel:`Open Settings`.
      #. On the ActivityWatch web page, scroll down to the :guilabel:`Server Settings` section and
         paste the Odoo database URL in the :guilabel:`CORS` field.
      #. Click :guilabel:`Next` twice.
      #. Click :guilabel:`Test connection` to verify the connection between the Odoo database and
         the ActivityWatch server.
```

   .. tab:: Mac

>       #. Install ActivityWatch

Reviewer: I'd remove this step completely and just have the sub-list items as main list items, like this:

<img width="877" height="187" alt="Image" src="https://github.com/user-attachments/assets/3e26c5c7-380c-4cd5-93b1-4d285134a7a4" />

         #. Odoo automatically detects that your operating system is macOS and offers you the option
            to download the appropriate ActivityWatch installation package for your configuration by
            clicking :guilabel:`Download for macOS`. You can also download the `macOS

>             installation package <https://github.com/ActivityWatch/activitywatch/releases/download/v0.13.2/activitywatch-v0.13.2-macos-x86_64.dmg>`_.

Reviewer: ```suggestion
         #. `Download the ActivityWatch package for macOS
             <https://github.com/ActivityWatch/activitywatch/releases/download/v0.13.2/activitywatch-v0.13.2-macos-x86_64.dmg>`_.
```
Since we can't install ActivityWatch on a Mac from Odoo yet

         #. Once downloaded, open the .dmg file and move the ActivityWatch application into your
            Mac's Applications folder.
         #. Go to the Applications folder and launch ActivityWatch.
         #. Click on the the ActivityWatch icon that is now in the menu bar and :

           - Click :guilabel:`Open Dashboard` to see a timeline of events captured by ActivityWatch.
           - Click :guilabel:`Modules Menu` to display a list of watchers and servers:

             - Python Server (aw-server) – Default server.

>              - Rust Server (aw-server-rust) – More efficient, will eventually replace Python server.

Reviewer: ```suggestion
    #. Open the downloaded `.dmg` file and move the ActivityWatch app into your
         :guilabel:`Applications` folder.
     #. Launch ActivityWatch.
     #. Click the ActivityWatch icon in the menu bar, select :guilabel:`Modules Menu`, and
         enable one of the following options:

         - :guilabel:`Python Server (aw-server)` (default server)
         - :guilabel:`Rust Server (aw-server-rust)` (more efficient server)
```

- I don't think the part with Open dashboard is a necessary step for the installation; I strongly suspect this displays the same data as when you enable debug mode in the timesheet assistant view in odoo and click View ActivityWatch data (which I made sure to mention in the use section).
- And just a note make sure to correct the indentation for the two servers; I think there's a space missing in your original text

> 

Reviewer: Since you configure either one server or the other, I wouldn't have both as steps (numbered list items) because that usually implies you do one step then move on to the next; instead, I'd have the following suggestion as a step and then use bold for each different server (not ideal, but this whole configuration is already a bit crazy so I don't think we are à ça près 😅 ):

```suggestion
#. Click the ActivityWatch icon in the menu bar and select :guilabel:`Open Configuration Folder.`
#. Configure the server based on the option selected earlier.
```
(I suggested to move the step about opening the configuration server here since it's common to both servers, to avoid repetitions and clarify the process as much as possible)

      #. Configure Python Server (aw-server)

         Skip this section if :guilabel:`aw-server-rust` is enabled and go to Rust Server
         Configuration below.

         - Click the ActivityWatch icon in the menu bar and go to
           :guilabel:`Open Configuration Folder.`
         - The Finder opens the configuration folder, which contains:

           - :guilabel:`aw-client`: Web interface for user settings and timeline.
           - :guilabel:`aw-qt`: Main program launched by ActivityWatch.
           - :guilabel:`aw-server`: Python server configuration folder.
           - :guilabel:`aw-server-rust`: Rust server configuration folder.
           - :guilabel:`Watcher folders`: Programs capturing activity and sending it to the server.

>          - Open :guilabel:`aw-server/aw-server.toml`.

Reviewer: ```suggestion
         **Python server configuration**

         #. In the :guilabel:`activitywatch` folder, open the :guilabel:`aw-server` folder, then the :file:`aw-server.toml` file.
```
- I don't think we need to explain what each folder does at this point since we're in the installation instructions I suggest we focus on what is needed for install only.
- I'd recommend using the [file markup](https://www.odoo.com/documentation/master/contributing/documentation/rst_guidelines.html#file) for the file name and name the folders individually

         - To allow your Odoo database to communicate with ActivityWatch, update the CORS parameters
           to match this:

            [server]

           #host = "localhost"

           #port = "5600"

           #storage = "peewee"

           cors_origins = `"https://example.odoo.com"`

           [server.custom_static]

           [server-testing]

           #host = "localhost"

           #port = "5666"

           #storage = "peewee"
           #cors_origins = ""

           [server-testing.custom_static]

            .. important::
               In the parameters above, make sure to replace `https://example.odoo.com` with the URL
               of your odoo instance, in quotes and without a slash (/) at the end.

>          - Save the file.

Reviewer: ```suggestion

         #. In the `[server]` section, update the `#cors_origins` parameter with your Odoo database URL, without a trailing slash:

            .. code-block:: toml

               [server]
               #host = "localhost"
               #port = "5600"
               #storage = "peewee"
               #cors_origins = "https://your-odoo-database-url.com"

         #. Leave the remaining parameters unchanged and save the file.
```

- I'd use a [code block](https://www.odoo.com/documentation/master/contributing/documentation/rst_guidelines.html#code-blocks)
- Since we actually only have to change one line in the file, I don't think we need to copy the rest of the code? I suggest including only the section of the code that needs to be changed and then mentioning that the rest can stay as is.
- I'd replace the URL used in the code so it's a bit more obvious that people need to replace it
- I think you forgot the # before `cors_origins`
- You'll probably need to rework the indentations in my suggestion a bit if you copy and paste it

Here's what it would look like:

<img width="1078" height="413" alt="Image" src="https://github.com/user-attachments/assets/f7e33ff4-2a54-429f-bec2-95be2646b9a9" />

      #. Configure Rust Server (aw-server-rust)

         - Click the ActivityWatch menu in the menu bar and go to
           :guilabel:`Open Configuration Folder`.
         - Open :guilabel:`aw-server-rust/config.toml`.
         - To allow your Odoo database to communicate with ActivityWatch, update the CORS parameters
           to match this:

            ### DEFAULT SETTINGS ###

           #address = "127.0.0.1"

           #port = 5600

           cors = [`"https://example.odoo.com"`]

           #

           #[custom_static]

           .. important::
               In the parameters above, make sure to replace `https://example.odoo.com` with the URL
               of your odoo instance, in quotes and without a slash (/) at the end.

>          - Save the file.

Reviewer: Same as above (but I also think you added backticks for the URL name (which the Google doc doesn't mention)
```suggestion
         **Rust server configuration**

         #. In the :guilabel:`activitywatch` folder, open the :guilabel:`aw-server-rust` folder,
            then the :file:`config.toml` file.
         #. In the `DEFAULT SETTINGS` section, update the `#cors` parameter with your Odoo database
            URL, without a trailing slash:

            .. code-block:: toml

               ### DEFAULT SETTINGS ###
               #address = "127.0.0.1"
               #port = 5600
               #cors = ["https://your-odoo-database-url.com"]
               #
               #[custom_static]

         #. Leave the remaining parameters unchanged and save the file.
```

      #. Restart ActivityWatch

         - To apply configuration changes:

           - Click the :guilabel:`ActivityWatch` menu and :guilabel:`Quit ActivityWatch`.

>            - Relaunch :guilabel:`ActivityWatch` from the Applications folder.

Reviewer: Unless this is specific to this app, I would just say to restart the app?
```suggestion
      #. Restart the ActivityWatch app.
```

         Timesheets Assistant to continue setup : :menuselection:`Timesheets --> Assistant --> Start
         Now`

      #. To help Timesheets Assistant automatically suggest your timesheet entries, your browser needs
         to be able to communicate with your local ActivityWatch server. Click
         :guilabel:`Open Settings` to access the ActivityWatch website. Under Server Settings,
         locate the CORS field. Copy and paste your Odoo database URL exactly as shown, in quotes
         and without a trailing slash (/) . Example: `"https://my-company.odoo.com"`

      #. Click the Odoo icon in your system tray (usually in the top bar or bottom right of your
         screen), then click :guilabel:`Restart Server`.

      #. Your browser may block connections to apps or services running on your computer. If
         prompted, allow access so the Assistant can detect and connect to ActivityWatch. Most
         browsers will prompt automatically when needed.

      #. Timesheets Assistant is now ready to track your activity. Click :guilabel:`Test Connection`
         to verify that everything is working. Start the server from the system tray at the

>          beginning of your work day.

Reviewer: - Mostly same comments as for Windows
- I had "Start server" on my computer, not "Restart server"; I assume this might depend on the OS so I added both in the instructions just in case.
- Since we already allowed Apps on device earlier, I don't think we need to do it again so I removed that step from the instruction

```suggestion
      #. In the :guilabel:`Install Timesheets Assistant` window, click :guilabel:`Download for Linux` in the :guilabel:`Desktop Activity` section, then select your OS version.

        .. note::
           The `Linux installation package can also be downloaded directly
           <https://nightly.odoo.com/aw/ubuntu/activitywatch-odoo.deb>`_.

      #. Install the package and follow the instructions.
      #. Once the installation is finished, restart your computer or log out and log back in.
      #. Access the Odoo database, go to :menuslection:`Timesheets --> Assistant`, and click
         :guilabel:`Start Now`.
      #. Click :guilabel:`Next` twice.
      #. In the :guilabel:`Configure CORS` window, copy the Odoo database URL.
      #. Click :guilabel:`Open Settings`.
      #. On the ActivityWatch web page, scroll down to the :guilabel:`Server Settings` section and
         paste the Odoo database URL in the :guilabel:`CORS` field.
      #. Click :guilabel:`Next`.
      #. Click the Odoo icon in your system tray (usually in the top bar or bottom right of your
         screen), then click :guilabel:`Start Server` or :guilabel:`Restart Server`.
      #. Click :guilabel:`Next` twice.
      #. Click :guilabel:`Test connection` to verify the connection between the Odoo database and
         the ActivityWatch server.
```

.. note::

   Optionally, ActivityWatch also provides
   `a wide range of watchers <https://docs.activitywatch.net/en/latest/watchers.html>`_ dedicated to
   specific activities that users can manually install if they want. Using more activity watchers

>    allows Timesheets Assistant to provide more accurate results.

Reviewer: ```suggestion
   specific activities that can be installed manually. Using more activity watchers allows the timesheet assistant to provide more accurate results.
```

Browser activity
~~~~~~~~~~~~~~~~

:guilabel:`Browser Activity` (recommended) tracks user activity within the browser based on the open
tabs and websites. Although optional, enabling Browser Activity allows Timesheets Assistant to

> generate more accurate results by processing additional data.

Reviewer: ```suggestion
Optionnally, install the ActivityWatch browser extension to track time spent in browser tabs and websites.

.. note::
   ActivityWtach browser extensions are only available for Google Chrome and Mozilla Firefox.
```

- I think the rest of the info in your original text is implied in the suggestion
- I think we should write that only Firefox and Chrome are supported for this (hence my suggestion to write a note about this)

Click on the Firefox or Google Chrome button to install the right extension for your browser, or
download it here for `Firefox <https://addons.mozilla.org/en-US/firefox/addon/aw-watcher-web/>`_ or

> `Google Chrome <https://chromewebstore.google.com/detail/activitywatch-web-watcher/nglaklhklhcoonedhgnpgddginnjdadi>`_.

Reviewer: ```suggestion
To do so, follow these steps:

#. Go to :menuselection:`Timesheets --> Assistant`.
#. Click the :icon:`fa-cog` (:guilabel:`Settings`) icon in the right part of the screen and select :guilabel:`Install Assistant`. 
#. In the :guilabel:`Install Timesheets Assistant` window, click the Firefox or Chrome extension button  in the :guilabel:`Browser activity` section, depending on your browser.
#. Click :guilabel:`Add to Chrome`, then :guilabel:`Add extension` for Chrome, or :guilabel:`Add to Firefox`, then :guilabel:`Add` for Firefox.

.. note::
   The `Firefox <https://addons.mozilla.org/en-US/firefox/addon/aw-watcher-web/>`_ and `Chrome <https://chromewebstore.google.com/detail/activitywatch-web-watcher/nglaklhklhcoonedhgnpgddginnjdadi>`_ extensions can also be installed from their respective store pages.
```
- to be more complete (hence the numbered list as well)
- once you've installed ActivityWatch for desktop activity, you don't see the start now button anymore so I think it's worth it to mention how we can access the install assistant window here
- for general style

> Assistant Rules

Reviewer: ```suggestion
Assistant rules
```

---------------

Assistant rules use regular expressions (regex) to provide Odoo with guidance on how to interpret
user activity tracked on their desktop and browser, and to convert it into timesheet entry

> suggestions.

Reviewer: ```suggestion
Assistant rules use regular expressions (regex) to define how Odoo interprets desktop and browser activity and converts it into timesheet entry suggestions.
```
more direct

Then, to be discussed, I would consider maybe adding the info that I suggested to take out from your general intro and move here:
**I'm not sure about some of the info I added, I'd like to check together!**

```suggestion
These rules act as *key events*: once a key event is triggered, any subsequent activity of at least one minute is associated with that event until a new key event is triggered by a matching rule. Preconfigured rules are available to track time spent in Odoo, GitHub, Google Docs, Gmail, Google Meet, and Discord.
```

.. note::

   Regular expressions, often abbreviated as regex, can be used in Odoo in various ways to search,
   validate, and manipulate data. Regex can be powerful but also complex, so it’s essential to use

>    it judiciously. `Learn more about Regex and how to use it <https://regexone.com/>`_.

Reviewer: ```suggestion
.. tip::
   - Regex can be powerful but complex. `RegexOne <https://regexone.com/>`_ provides interactive exercises to help learn it.
   - Adding more rules improves the precision of timesheet suggestions and reduces the impact of short or irrelevant entries.

```
- Great idea to add a link to a useful doc/tutorial on the topic
- You can remove the blank line under the note directive
- I suggested to remove the information that is repetitive (e.g., you already say that regular expressions =  regex in the paragraph above) or not really relevant or necessary in this context (e.g., that they can be used in odoo in various ways)
- I'd make this a tip instead of a note
- And then I suggested adding some of the info from your original intro to this tip here (I thought it was relevant)

To add a rule, go to :menuselection:`Timesheets --> Configuration --> Assistant Rules --> New`, then

> fill in the following fields:

Reviewer: ```suggestion
To create or edit an assistant rule, go to :menuselection:`Timesheets --> Configuration --> Assistant Rules`, then click :guilabel:`New` or select an existing rule, and fill in or edit the fields described below. Once done, click :guilabel:`Save`.
```
- Like you mentionned in your original intro, and rom what I tested in a trial db, the timesheet assistant comes with preconfigured rules out of the box so I think we should mention them here. And on that topic, I assume people might want to change the default rules so I think we should also include this in the text.
- only menus in menuselections
- It looks like we need to manually save so I think we should add this step as well

> - :guilabel:`Name`: The title you want to give to your rule.

Reviewer: here I think we can easily avoid using *you*:

```suggestion
- :guilabel:`Name`: The name of the rule.
```

- :guilabel:`Regex`: The pattern used to classify an activity as a key event by capturing
  ActivityWatch events (window name, tab name or URL) and display them in the Timesheet Assistant

>   when their duration exceeds 1 minute.

Reviewer: ```suggestion
- :guilabel:`Regex`: The pattern matched against ActivityWatch events (window name, tab name or URL) to identify and classify activities as key events and display them in the timesheet assistant.
```
- The original sentence made it unclear what the field actually does so I tried to make it a bit more logical (as much as I could, this is a bit complex for me)
- I think the part with the 1 minute minimum would be much more useful in the Use section (I added a comment there to add it)
- I think that it would be useful to have one or several examples here at some point. Let's keep this in my mind for the next update of this doc (when this feature has matured a bit maybe ...)

> - :guilabel:`Type`: Choose the icon displayed in the assistant suggestions list.

Reviewer: ```suggestion
- :guilabel:`Type`: Determines the icon displayed for matching entries in the :guilabel:`Suggestions` list of the timesheet assistant.
```
Because you don't really choose an icon (I was expecting to see a list of icons to choose from after reading this); instead, the type you select will determine the icon displayed

> - :guilabel:`Template`: Defines the name of the generated timesheet entry and is used by local rules

Reviewer: ```suggestion
- :guilabel:`Template`: Defines the name of the generated timesheet entry and is used by :ref:`local rules <timesheets/timesheets-assistant/local-rules>`
```
to provide more context (don't forget to add the anchor name I added in my suggestion above the Local assistant rules heading, on line 303)

>   for further processing. If left blank, the rule name is used instead.

Reviewer: ```suggestion
  for further processing. If left blank, the rule's :guilabel:`Name` is used instead.
```
since we're referring to the field

- :guilabel:`Description`: A display-only label shown in the Timesheet Assistant. It is purely

>   indicative and has no impact on local rules or timesheet generation.

Reviewer: I'm not sure I understand the difference between the template and the description based on your explanation here.

From what I tested and understood: The description will also be used to define the name of the generated timesheet entry if there is one. If there's no description, the template is used; if there's not template, the rule's name is used. I understand why we woud need a template but I don't understand why we need a template AND a description? In what exact case(s) would we need both?

Author: See my general comment about regex below.

- :guilabel:`Project` and :guilabel:`Task`: The values selected by default for activities matching

>   this rule. If left blank, local rules apply.

Reviewer: By default, those fields do not appear in the list, we have to click the sliders and then select the fields; I recommended moving these two fields to a tip after the Always active field description below, and rephrase this as follows for accuracy:

```suggestion
   .. tip::
      To set a default project and task for timesheet entries created from matching activities, click the :icon:`oi-settings-adjust` (:guilabel:`additional options`) button in the list header, enable :guilabel:`Project` and :guilabel:`Task`, then select the desired values from the fields in the list view.
```
And I don't understand what you mean by "local rules" in the last sentence (hence why I didn't include it in my suggestion). I thought you meant local assistant rules, but that doesn't really make sense: If no project or task are assigned to the rule, no local assistant rule will be created for that rule/key event, will it? Could you please clarify this?

Author: See my general comment about regex below.

- :guilabel:`Always active`: Enable it to track activities matched by this rule, even if
  ActivityWatch marks the user as inactive after three minutes of inactivity. For example, if a
  user has a meeting, inactivity on the keyboard and mouse might be interpreted as a break. If the
  rule covering meetings has Always Active enabled, ActivityWatch does not mark the user as

>   inactive during that activity.

Reviewer: I think we can simplify this a bit:

```suggestion
- :guilabel:`Always active`: When enabled, activities matched by this rule are tracked even if ActivityWatch marks the user as inactive after three minutes of keyboard and mouse inactivity. This is useful for rules covering activities such as meetings.
```

.. example::

   Here's an example for a Google Meet rule:

   - :guilabel:`Name`: Customer Meeting

>    - :guilabel:`Regex`: Meet - (.*)\|https://meet.google.com

Reviewer: ```suggestion
   - :guilabel:`Regex`: `meet - (.*)\|https://meet.google.com`
```
- removed the capital letter since it's not capitalized

<img width="341" height="46" alt="Image" src="https://github.com/user-attachments/assets/ea1ceb55-c397-475b-9e3a-8ebca2707db2" />

- added [literal formatting](https://www.odoo.com/documentation/master/contributing/documentation/rst_guidelines.html#technical-term-literal), otherwise not all characters are rendered here
- Would it be useful to briefly explain here what this regex does? What do you think?

Author: Good idea! I explained it in the first text block of the example. Let me know what you think.

>    - :guilabel:`Type`: Videocall

Reviewer: ```suggestion
   - :guilabel:`Type`: Video Conference
```
(couldn't find Videocall in the list)

   - :guilabel:`Template`: Customer Meeting ($1) (refers to the value captured by the first group in

>      the regex.)

Reviewer: ```suggestion
     the regex)
```

   - :guilabel:`Description`: Customer Meeting (abc-123) (static example to illustrate — in practice
     , this is an indicative version of the template shown in the suggestions list, which can be

>      used at the user’s convenience, e.g., "Customer Meeting (meet-id)")

Reviewer: - I don't think this should be explained in an example. Instead, the field's description above should be clear.
- I think it might be useul to change the actual example abc-123 to something a bit more realistic to make it clearer here as well?

```suggestion
   - :guilabel:`Description`: Customer Meeting (abc-123)
```

   - :guilabel:`Project` and :guilabel:`Task`: Left empty, since each meeting may belong to a
     different customer/task
   - :guilabel:`Always active`: Enabled

> Local Assistant Rules

Reviewer: ```suggestion
Local assistant rules
```
I don't think we need a dedicated section for this since it's display only. Instead, I'd recommend a tip at the bottom of the previous *Assistant rules* section; see my comment below for how I recommend we rephrase the info to fit a tip.

---------------------

.. important::

   Local Assistant Rules are only available in
   :doc:`Developer mode </applications/general/developer_mode>` (also called *Debug Mode*).

   Turn on :guilabel:`Developer Mode`, then go to
   :menuselection:`Timesheets --> Configuration --> Local Assistant Rules`.

:guilabel:`Local Assistant Rules` show the project and task assigned to each ActivityWatch key
event. It is available for informational purposes, to help understand why Odoo selects a project or
task for a key event, or why Odoo does not detect the correct project or task.

When suggested timesheet entries are matched with projects, Odoo learns from those selections to
improve future suggestions. The data shown here indicate how timesheet entry patterns are analyzed

> to help recommend relevant projects and tasks based on past behavior.

Reviewer: ```suggestion
.. tip::
   To view the list of key events detected by ActivityWatch and the project and task assigned to them based on the configured :ref:`assistant rules <timesheets/timesheets-assistant/assistant-rules>` and/or the project and task assigned to past suggested timesheet entries, activate :ref:`developer mode <developer-mode>`, then go to :menuselection:`Timesheets --> Configuration --> Local Assistant Rules`.
```
Comments on your original text:
- the important note is not needed; instead we can just add developer mode activation in the instructions directly and it's quite common
- you can use the ref to refer to the dev mode documentation instead of a doc link (it was created specifically for the dev mode doc because it needs to be referred to so often)

Comments on my suggestion:
- Move this tip at the bottom of the Assistant rules section
- I think some of the info here would be more useful/relevant in the Use section (this suggestion, and the one below for the Use section include this already)

Author: Can you double-check how I added this part, please? I'm a bit confused at this point :-/

> Using Timesheets Assistant

Reviewer: ```suggestion
Use
```

==========================

Once Timesheet Assistant is properly setup, go to :menuselection:`Timesheets --> Assistant`.  In the
:guilabel:`Suggestions` section, click on one or multiple :guilabel:`Unmatched` timesheet
suggestions.

Then select a project in the left panel to assign the suggestion(s). You can also add a
:guilabel:`Description`, select a :guilabel:`Task`, or adjust the recorded time if needed, mark it
as :guilabel:`Billable` if required, and click :guilabel:`Create`.

.. note::
   The time rounding configured in :menuselection:`Configuration --> Settings --> Time Rounding` is
   applied to the timesheets suggestions.

Hover over a suggestion and click :guilabel:`Remove` to delete it if needed.

To edit a timesheet entry you’ve already created, click the entry, make the necessary changes, and
click :guilabel:`Save`.

While timesheet suggestions remain private to the user, once a timesheet is created, the data

> becomes public and visible to other users in your organization.

Reviewer: ```suggestion
Once the timesheet assistant has been :ref:`configured
<timesheets/timesheets-assistant/configuration>`, computer and browser (if
:ref:`applicable <timesheets/timesheets-assistant/install-browser>`) activities are
automatically tracked at startup each day, and calendar events and planning in Odoo are checked. Timesheet entry suggestions are then generated from :ref:`key events <timesheets/timesheets-assistant/assistant-rules>` detected by ActivityWatch based on tracked activities when their duration exceeds one minute.

To view the suggestions, go to :menuselection:`Timesheets --> Assistant`. By default,
:guilabel:`Suggestions` are listed :guilabel:`By Project`; click
:guilabel:`Chronological` in the top-right of the panel to switch to chronological order.

To create a timesheet entry from one or multiple suggestions, click the relevant
suggestion(s) in the list, :ref:`edit the fields <timesheets/timesheets/log-manually>`
in the left pane if necessary, and click :guilabel:`Create`.

To delete a suggestion, hover over it and click :guilabel:`Remove`. To delete multiple
suggestions at once, select them in the list and click :guilabel:`Delete`.

To edit or delete an existing timesheet entry, click it in the left pane, make the
necessary changes, then click :guilabel:`Save` or click :guilabel:`Delete`.

.. note::
   - The :guilabel:`Time Rounding` settings apply to timesheet suggestions.
   - Three minutes or more without mouse or keyboard activity are interpreted as inactivity (i.e., a break), unless the :guilabel:`Always Active` option is enabled for a matching :ref:`rule <timesheets/timesheets-assistant/assistant-rules>`.
   - Timesheet suggestions are automatically assigned to projects and tasks based on the
     configured :ref:`assistant rules
     <timesheets/timesheets-assistant/assistant-rules>` and the project and task assigned
     to past suggestions in the timesheet assistant.
   - Timesheet suggestions remain private until timesheet entries are created from them.

.. tip::
   To view the data recorded by ActivityWatch, activate :ref:`developer mode
   <developer-mode>`, go to :menuselection:`Timesheets --> Assistant`, and click
   :guilabel:`View ActivityWatch Data` at the bottom of the suggestions list.
```

Comments about your original text:

- I don't think we need to describe the fields since this is "standard" timesheet logging; instead, I suggested to add a ref to the relevant section on the main Timsheets doc page
- I couldn't find Unmatched, but I assume it shows if there is no project/task assigned to the suggestion? but even then, I don't think it's relevant; any suggestion (even with a project/task assigned) will remain just a suggestion if we don't select it and create an entry from it, right?
- Some info is missing: Chronological button, create an entry from multiple suggestions and delete multiple suggestions
- I think the Time Rounding setting should be added to the main Timesheets doc page as it's not specific to the assistant and it's not documented anywhere else; I created as separate Odoo task for this

Comments on my suggestion:

- Don't forget to add this anchor above the Configuration section: .. _timesheets/timesheets-assistant/configuration
- People need to start the server everyday to "launch" ActivityWatch on their computer and start recording activities --> EDIT: since they plan to merge a task soon to remove the need for this, I rephrased my suggestion to take that future change into account already
- I suggested to refer to the assistant rules section so we don't need to explain key events here again
- Moved relevant info from other sections (mainly, the general intro + local assistant rules section)
I think it'd be useful to mention the Always Active feature here as well with a ref

Author: _I couldn't find Unmatched, but I assume it shows if there is no project/task assigned to the suggestion? but even then, I don't think it's relevant; any suggestion (even with a project/task assigned) will remain just a suggestion if we don't select it and create an entry from it, right?_

Yes that's my understanding of it.
